sdk/src/azure/core/az_mqtt5_telemetry_consumer_hfsm.c

```diff
@@ -60,21 +60,9 @@ static az_result root(az_event_policy* me, az_event event)
 
     case AZ_HFSM_EVENT_ERROR:
     {
-      az_mqtt5_telemetry_consumer* this_policy = (az_mqtt5_telemetry_consumer*)me;
-      if ((az_event_policy*)me->inbound_policy != NULL)
+      if (az_result_failed(az_event_policy_send_inbound_event(me, event)))
```

Comment:
add handling (to properly ignore) for AZ_EVENT_MQTT5_CONNECTION_CLOSED_IND and AZ_EVENT_MQTT5_CONNECTION_OPEN_IND for the two telemetry hfsms like you did for the rpc hfsms